addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -381,11 +383,13 @@ def _get_component_data(self, parent_bom, parent_product, warehouse, bom_line, l
 
     @api.model
     def _get_quantities_info(self, product, bom_uom, product_info, parent_bom=False, parent_product=False):
-        return {
+        quantities_info = {
             'free_qty': max(product.uom_id._compute_quantity(product.free_qty, bom_uom), 0) if product.detailed_type == 'product' else 0,
             'on_hand_qty': product.uom_id._compute_quantity(product.qty_available, bom_uom) if product.detailed_type == 'product' else 0,
             'stock_loc': 'in_stock',
         }
+        quantities_info['free_to_manufacture_qty'] = quantities_info['free_qty']
```

Comment:
I think it's useless to have another key, every thing could be computed when needed in the `free_qty`

Reply:
@svs-odoo Thanks for the review. Adding the key `free_to_manufacture_qty` is important here because we need to make a distinction when we have a subcontracted BoM. The report needs to show the free quantity of BoM lines which is equal to the sum of products stock in the warehouse and products stock in subcontracting location. However, calculating producible quantity is different, because it is calculated based on the subcontracting location stock if this is a subcontracted BoM, otherwise it will be calculated base on the `free_qty` value. Therefore, if we have a subcontracted BoM, we call `_get_quantities_info()` in the base report module, then we update `free_to_manufacture_qty` in the overriden method in the `mrp_subcontracting` module.